distribution/lib/Standard/Database/0.0.0-dev/src/Data/Table.enso

```diff
@@ -477,21 +487,35 @@ type Table
 
        If a column with the given name already exists, it will be replaced.
        Otherwise a new column is added.
-    set : Text -> Column -> Table
-    set self name column = case Helpers.ensure_name_is_sane name of
-        True ->
-            is_used_in_index = self.context.meta_index.exists i-> i.name == name
-            case is_used_in_index of
-                True -> Error.throw <| Illegal_State_Error_Data "Cannot override column "+name+", because it is used as an index. Remove the index or use a different name."
-                False ->
-                    new_col = Internal_Column.Value name column.sql_type column.expression
-                    replace = self.internal_columns.exists (c -> c.name == name)
-                    case replace of
-                        True ->
-                            new_cols = self.internal_columns.map (c -> if c.name == name then new_col else c)
-                            self.updated_columns new_cols
-                        False ->
-                            self.updated_columns (self.internal_columns + [new_col])
+    set : Text -> Column | Text -> Problem_Behavior -> Table
+    set self name column on_problems=Report_Warning = on_problems.handle_errors fallback=self <|
+        case Helpers.ensure_name_is_sane name of
+            True ->
+                is_used_in_index = self.context.meta_index.exists i-> i.name == name
+                case is_used_in_index of
+                    True -> Error.throw <| Illegal_State_Error_Data "Cannot override column "+name+", because it is used as an index. Remove the index or use a different name."
+                    False ->
+                        resolved = case column of
+                            _ : Text -> self.evaluate column
+                            _ -> column
+                        new_col = Internal_Column.Value name resolved.sql_type resolved.expression
+                        replace = self.internal_columns.exists (c -> c.name == name)
+                        case replace of
+                            True ->
+                                new_cols = self.internal_columns.map (c -> if c.name == name then new_col else c)
+                                self.updated_columns new_cols
+                            False ->
+                                self.updated_columns (self.internal_columns + [new_col])
+
+    ## PRIVATE
+    evaluate : Text -> Column
+    evaluate self expression =
+        getColumn name = self.at name
+        makeConstant value =
+            new_type = SQL_Type.approximate_type value
+            other = Expression.Constant new_type value
+            Column.Value (UUID.randomUUID.to_text) self.connection new_type other self.context
```

Comment:
```suggestion
            Column.Value ("Constant_"+UUID.randomUUID.to_text) self.connection new_type other self.context
```
I'm not sure but maybe a prefix explaining what this column _is_ before the random string would make it a bit easier to understand a generated expression while debugging.